### ISSUE TITLE
ethereum network

### ISSUE THREAD

**Issue by ramin1234** [bitlogik/lattice-attack#10]
does it work it ethereum network?

**Comment by bitlogik** [comment]
Yes it does. It can work with any system where you have / you know :

- ECDSA (with "SECP224R1", "SECP256K1", "SECP256R1", "SECP384R1" or "SECP521R1" curve)
- A public key
- A set of signatures done by this key
- The partial information about the _ephemeral random internal secret nonce_ (from a side channel, or "internal" info) for each of these signatures.

As soon as you get a set of transactions from the Ethereum blockchain (for a given address), and with their corresponding ephemeral random internal secret ECDSA number partial info (MSB or LSB), you can use lattice-attack to recover the private key. Of course this requires some work to process the transactions data to decode the signature into the format input of this software. That needs also to recover the public key from signature as per Ethereum design.
